posydon/unit_tests/utils/test_common_functions.py

```diff
@@ -556,6 +560,28 @@ def test_roche_lobe_radius(self, capsys):
             assert totest.roche_lobe_radius(m, 1.0)\
                    + totest.roche_lobe_radius(1.0, m) < 1.0
 
+    def test_check_for_RLO(self):
+        # missing argument
+        with raises(TypeError, match="missing 5 required positional "\
+                                     +"arguments: 'm1', 'r1', 'm2', 'r2', "\
+                                     +"and 'separation'"):
+            totest.check_for_RLO()
+
+        # examples
+        tests = [(1.0, 1.0, 1.0, 1.0, 1.0, 1e-3, True),
+                 (2.0, 1.0, 1.0, 1.0, 2.5, 1e-3, True),
+                 (1.0, 1.0, 2.0, 1.0, 2.5, 1e-3, True),
+                 (1.0, 1.0, 1.0, 1.0, 4.0, 1e-3, False),
+                 (1.0, 1.0, 2.0, 1.0, 2.5, 1e-8, True)]
+
+        for (m1, r1, m2, r2, separation, tolerance, RLO) in tests:
+            if tolerance == 1e-8:
+                assert totest.check_for_RLO(m1, r1, m2, r2, separation) == RLO
+            else:
+                assert totest.check_for_RLO(m1, r1, m2, r2, separation,\
+                                            tolerance) == RLO
+                
```

Comment:
The new test case is only testing, whether you need to give the tolerance, while it looks like testing the effect of the tolerance. Thus, I suggest:
```suggestion
        tests = [(1.0, 1.0, 1.0, 1.0, 1.0, 1e-3, True),
                 (2.0, 1.0, 1.0, 1.0, 2.5, 1e-3, True),
                 (1.0, 1.0, 2.0, 1.0, 2.5, 1e-3, True),
                 (1.0, 1.0, 1.0, 1.0, 4.0, 1e-3, False)]

        for (m1, r1, m2, r2, separation, tolerance, RLO) in tests:
            assert totest.check_for_RLO(m1, r1, m2, r2, separation) == RLO
            assert totest.check_for_RLO(m1, r1, m2, r2, separation, tolerance)\
                   == RLO
```

Reply:
In case you like to check the tolerance effect, you could add a test like
`(1.0, 1.0, 1.0, 1.0, 1.0, 9e-1, False)`